src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -547,9 +548,13 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
                 (sObj.get('CampaignId') != null && mapWrapper.containsKey((Id)sObj.get('CampaignId'))))) {
 
                 alloWrapper wrap;
+                Map<Id, Allocation__c> GAUtoAlloMapping;
                 //get wrapper for parent Payment.
-                if (sObj instanceof npe01__OppPayment__c)
+                if (sObj instanceof npe01__OppPayment__c) {
                     wrap = mapWrapper.get((Id)sObj.get('npe01__Opportunity__c'));
+                    GAUtoAlloMapping = new Map<Id, Allocation__c>();
```

Comment:
This would be easier to follow (further down), if you set an indicator like isPaymentAllocation.  Checking for null/not null for the GAUtoAlloMapping made the intent less clear.

Reply:
Would also be fine with the check sObj instanceof npe01__OppPayment__c in all cases, as this is clearer.